nat-lab/tests/utils/testing.py

```diff
@@ -10,8 +12,48 @@ def unpack_optional(opt: Optional[T]) -> T:
     return opt
 
 
-def test_name_safe_for_file_name():
+def get_current_test_case_and_parameters() -> Tuple[Optional[str], Optional[str]]:
+    """
+    Returns a tuple of strings representing the case name and parameters of the currently running test.
+
+    The result is derived from PYTEST_CURRENT_TEST env variable.
+    """
     test_name = os.environ.get("PYTEST_CURRENT_TEST")
-    if test_name is not None:
-        return "".join([x if x.isalnum() else "_" for x in test_name.split(" ")[0]])
-    return test_name
+    if test_name:
+        test_parts = test_name.split("::", 1)[-1].split(" ")[0].split("[", 1)
+        if len(test_parts) > 1:
+            return (
+                format_path_string(test_parts[0]),
+                format_path_string(test_parts[1]),
+            )
+        return (format_path_string(test_parts[0]), None)
+    print(datetime.now(), "PYTEST_CURRENT_TEST is None")
+    return (None, None)
+
+
+def format_path_string(input_str: str) -> str:
```

Comment:
This is now "private" to this file. We seem to have a practice of prepending those with _